python/cudf/cudf/core/frame.py

```diff
@@ -3594,6 +3594,64 @@ def _colwise_binop(
 
         return output
 
+    def dot(self, other, reflect=False):
+        """
+        Get dot product of frame and other, (binary operator `dot`).
+
+        Among flexible wrappers (`add`, `sub`, `mul`, `div`, `mod`, `pow`,
+        `dot`) to arithmetic operators: `+`, `-`, `*`, `/`, `//`, `%`, `**`,
+        `@`.
+
+        Parameters
+        ----------
+        other : sequence, Series, or DataFrame
+            Any multiple element data structure, or list-like object.
+        reflect : bool, default False
+            If ``True`` the operation is reflected (i.e whether to swap the
```

Comment:
In the context of the binary ops, this indicates that `other` is on the left hand side instead of the right hand side; in cases where we are doing dot products using the `@` binary operator, this would be necessary if our LHS doesn't implement `__matmul__()`. Essentially

```python
cudf_frame_object @ [1, 2, 3, 4]  # calls Frame.__matmul__() -> Frame.dot(...reflect=False)
[1, 2, 3, 4] @ cudf_frame_object  # calls Frame.__rmatmul__() -> Frame.dot(...reflect=True)
```

It would probably be helpful to clarify this a bit better in the docstrings.

Reply:
thanks for the explanation, yes i also think you should clarify that further in the docstrings :))